catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/ResourceOperations.java

```diff
@@ -0,0 +1,750 @@
+        Map<String, Set<String>> optionsMap;
+        try {
+            LOGGER.debug("source id to get options from: {}", sourceId);
+            QueryRequest queryRequest = new QueryRequestImpl(createMetacardIdQuery(metacardId),
+                    false,
+                    Collections.singletonList(sourceId == null ? this.getId() : sourceId),
+                    null);
+            QueryResponse queryResponse = queryOperations.query(queryRequest,
+                    null,
+                    false,
+                    fanoutEnabled);
+            List<Result> results = queryResponse.getResults();
+
+            if (!results.isEmpty()) {
+                Metacard metacard = results.get(0)
+                        .getMetacard();
+                // DDF-1763: Check if the source ID passed in is null, empty,
+                // or the local provider.
+                if (StringUtils.isEmpty(sourceId) || sourceId.equals(getId())) {
+                    optionsMap = Collections.singletonMap(ResourceRequest.OPTION_ARGUMENT,
+                            getOptionsFromLocalProvider(metacard));
+                } else {
+                    optionsMap = Collections.singletonMap(ResourceRequest.OPTION_ARGUMENT,
+                            getOptionsFromFederatedSource(metacard, sourceId));
+                }
+            } else {
+
+                String message = "Could not find metacard " + metacardId + " on source " + sourceId;
+                throw new ResourceNotFoundException(message);
+            }
+        } catch (UnsupportedQueryException e) {
+            LOGGER.warn("Error finding metacard {}", metacardId, e);
+            throw new ResourceNotFoundException("Error finding metacard due to Unsuppported Query",
+                    e);
+        } catch (FederationException e) {
+            LOGGER.warn("Error federating query for metacard {}", metacardId, e);
+            throw new ResourceNotFoundException("Error finding metacard due to Federation issue",
+                    e);
+        } catch (IllegalArgumentException e) {
+            LOGGER.warn("Metacard couldn't be found {}", metacardId, e);
+            throw new ResourceNotFoundException("Query returned null metacard", e);
+        } finally {
+            LOGGER.trace("EXITING: getResourceOptions");
+        }
+
+        return optionsMap;
+    }
+
+    //
+    //
+    //
+    @SuppressWarnings("javadoc")
+    public ResourceResponse getResource(ResourceRequest resourceRequest, boolean isEnterprise,
+            String resourceSiteName, boolean fanoutEnabled)
+            throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
+        ResourceResponse resourceResponse = null;
+        ResourceRequest resourceReq = resourceRequest;
+        String resourceSourceName = resourceSiteName;
+        ResourceRetriever retriever = null;
+
+        if (fanoutEnabled) {
+            isEnterprise = true;
```

Comment:
this is a little weird you don't actually need isEnterprise since it will always be equivalent to fanoutEnabled?